biom/parse.py

```diff
@@ -245,6 +246,98 @@ def get_axis_indices(biom_str, to_keep, axis):
+        "post-split-libraries" format, where the identifiers are of the form
+        ``<sample-id>_<sequence-id>``. Everything before the first underscore
+        will be used as the sample identifier in the resulting ``Table``.
+        The information after the first underscore is not used directly, though
+        the full identifiers of seeds will be used as the observation
+        identifier in the resulting ``Table``.
+
+    """
+    data = collections.defaultdict(int)
+    sample_idxs = {}
+    sample_ids = []
+    observation_idxs = {}
+    observation_ids = []
+    # The types of hit lines we need here are hit (H), seed (S) and
+    # library seed (L). Store these in a set for quick reference.
+    line_types = set(list('HSL'))
+    for line in fh:
+        # determine if the current line is one that we need
+        line = line.strip()
+        if not (line and line[0] in line_types):
+            # if the line is blank, or not one of the types we need,
+            # including comments, move on to the next line
+            continue
+        else:
+            fields = line.split('\t')
+
+        # grab the fields we care about
+        hit_type = fields[0]
+        observation_id = fields[9].split()[0]
+        query_id = fields[8].split()[0]
+
+        if observation_id == '*':
+            # S and L lines don't have a separate observation id
+            observation_id = query_id
+
+        # get the index of the current observation id, or create it if it's
+        # the first time we're seeing this id
+        try:
+            observation_idx = observation_idxs[observation_id]
+        except KeyError:
+            observation_idx = len(observation_ids)
+            observation_ids.append(observation_id)
+            observation_idxs[observation_id] = observation_idx
+
+        if hit_type == 'H' or hit_type == 'S':
+            # get the sample id
+            try:
+                underscore_index = query_id.index('_')
+            except ValueError:
+                raise ValueError(
+                 "A query sequence was encountered that does not have an "
+                 "underscore. An underscore is required in all query "
+                 "sequence identifiers to indicate the sample identifier.")
+            # get the sample id and its index, creating the index if it is the
+            # first time we're seeing this id
+            sample_id = query_id[:underscore_index]
+            try:
+                sample_idx = sample_idxs[sample_id]
+            except KeyError:
+                sample_idx = len(sample_ids)
+                sample_idxs[sample_id] = sample_idx
+                sample_ids.append(sample_id)
+            # increment the count of the current observation in the current
+            # sample by one.
+            data[(observation_idx, sample_idx)] += 1
+        else:
+            # nothing else needs to be done for 'L' records
```

Comment:
would it be possible to have an explicit branch for this, and raise in `else`? That would add some protection against corrupt or unexpected input